src/gflags.cc

```diff
@@ -800,7 +826,7 @@ CommandLineFlag* FlagRegistry::SplitArgumentLocked(const char* arg,
                                     kError, key->c_str());
       return NULL;
     }
-    if (strcmp(flag->type_name(), "bool") != 0) {
+    if (!flag->OfType<bool>()) {
```

Comment:
On the other hand, does `FlagValue::OfType<T>` really add to the `.o` file size? Would it not just be inlined by the compiler as if you wrote

```c++
if (flag->type_ == FlagValueTraits<bool>:kValueType)
```

and the `FlagValueTraits` definition is anyway in the `.o`? I still think a `ValueType FlagValue::Type() const` function would be sufficient, but I guess it's more a matter of taste rather than `.o` file size.